pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00_test.go

```diff
@@ -393,8 +393,17 @@ func TestAvailableStatus(t *testing.T) {
 			if condition.Status != tc.expectedStatus {
 				t.Error(diff.ObjectGoPrintSideBySide(condition.Status, tc.expectedStatus))
 			}
-			if tc.expectedReason != "" && condition.Reason != tc.expectedReason {
-				t.Error(diff.ObjectGoPrintSideBySide(condition.Reason, tc.expectedReason))
+			if len(tc.expectedReasons) > 0 {
+				actualReasons := strings.Split(condition.Reason, "\n")
+				a := make([]string, len(tc.expectedReasons))
+				b := make([]string, len(actualReasons))
+				copy(a, tc.expectedReasons)
+				copy(b, actualReasons)
+				sort.Strings(a)
+				sort.Strings(b)
+				if !equality.Semantic.DeepEqual(a, b) {
+					t.Error("\n" + diff.ObjectDiff(a, b))
+				}
```

Comment:
Why not to compare this as strings instead? You know what those should look like

Reply:
I think I understood what's missing here. I should have used sort function only `expectedReasons` array because the `actualReasons` array would already be in sorted order.